gammapy/irf/core.py

```diff
@@ -1,17 +1,284 @@
+        """
+        # TODO: change to coord dict?
+        non_valid_axis = set(kwargs).difference(self.axes.names)
+        if non_valid_axis:
+            raise ValueError(
+                f"Not a valid coordinate axis {non_valid_axis}"
+                f" Choose from: {self.axes.names}"
+            )
+
+        coords_default = self.axes.get_coord()
+
+        for key, value in kwargs.items():
+            coord = kwargs.get(key, value)
+            if coord is not None:
+                coords_default[key] = u.Quantity(coord, copy=False)
+
+        return self._interpolate(coords_default.values(), method=method)
+
+    # TODO: define a proper integration method
+    def integrate_energy(self, method="linear", **kwargs):
+        """Integrate in a given energy band.
+
+        This method uses log-log trapezoidal integration.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Coordinates at which to evaluate the IRF
+        method : {'linear', 'nearest'}, optional
+            Interpolation method
 
+        Returns
+        -------
+        array : `~astropy.units.Quantity`
+            Returns 2D array with axes offset
+        """
+        axis = self.axes.index("energy")
+        data = self.evaluate(**kwargs, method=method)
+        energy = kwargs["energy"]
+        return trapz_loglog(data, energy, axis=axis)
+
+    # TODO: define a proper integration method
+    @lazyproperty
+    def _integrate_rad(self):
+        rad_axis = self.axes["rad"]
+        rad_drad = (
+                2 * np.pi * rad_axis.center * self.quantity * rad_axis.bin_width
+        )
+        idx_rad = self.axes.index("rad")
+        values = rad_drad.cumsum(axis=idx_rad).to_value("")
+        values = np.insert(values, 0, 0, axis=idx_rad)
+
+        points = [ax.center for ax in self.axes]
+        points[idx_rad] = rad_axis.edges
+        return ScaledRegularGridInterpolator(
+            points=points, values=values, fill_value=1,
+        )
+
+    @classmethod
+    def from_hdulist(cls, hdulist, hdu=None):
+        """Create from `~astropy.io.fits.HDUList`.
+
+        Parameters
+        ----------
+        hdulist : `~astropy.io.HDUList`
+            HDU list
+        hdu : str
+            HDU name
+
+        Returns
+        -------
+        irf : `IRF`
+            IRF class
+        """
+        if hdu is None:
+            hdu = IRF_DL3_HDU_SPECIFICATION[cls.tag]["hdu"]
+
+        return cls.from_table(Table.read(hdulist[hdu]))
+
+    @classmethod
+    def read(cls, filename, hdu=None):
+        """Read from file.
+
+        Parameters
+        ----------
+        filename : str or `Path`
+            Filename
+        hdu : str
+            HDU name
+
+        Returns
+        -------
+        irf : `IRF`
+            IRF class
+        """
+        with fits.open(str(make_path(filename)), memmap=False) as hdulist:
+            return cls.from_hdulist(hdulist, hdu=hdu)
+
+    @classmethod
```

Comment:
I think to prepare the likely multiplicity of formats we need to provide the `format` as an argument to read/write and `from_table` methods.

Reply:
Right now there is only the `gadf-dl3` format and we will not get any other format soon, but I agree we could prepare the API for this...